app/controllers/questions_controller.rb

```diff
@@ -29,11 +29,26 @@ def show
     @answer = @question.answers.new
   end
 
+  def edit
+    @question = Question.find(params[:id])
+    @categories = Category.all
+  end
+
+  def update
+    @question = Question.find(params[:id])
+    if @question.update(question_params)
+      flash[:success] = "Your question was edited successfully!"
+      redirect_to question_path(@question)
+    else
+      flash[:danger] = "Failed to edit question. Please try again."
```

Comment:
We're going to eventually need to add in a targeted reason here as to why the update failed. Same goes for all sad paths in other controllers.